<launch>
    <node name="controller" pkg="pid" type="controller" ns="pid_ximg">
      <param name="Kp" value="-2.0" />
      <param name="Ki" value="0" />
      <param name="Kd" value="-1.5" />
      <param name="upper_limit" value="2.00" />
      <param name="lower_limit" value="-2.00" />
      <param name="windup_limit" value="0.4" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <!-- <remap from="state" to="/pid_roll/setpoint" /> -->
      <remap from="setpoint" to="/pid_zero_setpoint" />
     </node>

     <node name="controller" pkg="pid" type="controller" ns="pid_yimg">
      <param name="Kp" value="-2.0" />
      <param name="Ki" value="0" />
      <param name="Kd" value="-1.5" />
      <param name="upper_limit" value="2.00" />
      <param name="lower_limit" value="-2.00" />
      <param name="windup_limit" value="0.4" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <!-- <remap from="state" to="/pid_pitch/setpoint" /> -->
      <remap from="setpoint" to="/pid_zero_setpoint" />
    </node>

    <node name="vision_based_controller_mission" pkg="drone_controller" type="vision_based_mission.py" output="screen">
      <remap from="/err_x_img"   to="/pid_ximg/state"/>
      <remap from="/err_y_img"   to="/pid_yimg/state"/>
    </node>

    <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot" ns="vision_based_controller"
    args="/pid_ximg/control_effort
          /pid_yimg/control_effort" />

</launch>
